### ISSUE TITLE
include_role: inside a block: does not work with when: (even for non-string condition) 

### ISSUE THREAD

**Issue by joaocc** [ansible/ansible#19472]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include_role:

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT
N/A

##### SUMMARY
include_role: inside a block: does not work with when: crashes when starting to execute the included role.
If I remove the external when: clause, everything works fine.

(this is a re-submition of #19085 which was incorrectly closed)

##### STEPS TO REPRODUCE

calling script:
```
- block:
    - include_role:
          name: "ansible-lib--infra/v1/ez123--moodle-moosh"
          static:  no
          private: yes
      when: (zzz_moodle__skip__moosh is not defined) or (not (zzz_moodle__skip__moosh |bool))
```
on the included role, under tasks/main.yml
```
- name: "check if moosh is already installed (1)"
  command: "./moosh --help"
  args:
    chdir: "{{ ez123_moosh__bin_dir }}"
  register: _X_moosh_ver_0
  changed_when: false
  failed_when: false
```
##### EXPECTED RESULTS
There should be no error/crash.

##### ACTUAL RESULTS
```
TASK [ansible-lib--infra/v1/ez123--moodle-moosh : check if moosh is already installed (1)] ***
task path: /etc/ansible/roles.glx/ansible-lib--infra/v1/ez123--moodle-moosh/tasks/main.yml:4
fatal: [host000000]: FAILED! => {
    "failed": true,
    "msg": "The conditional check '{' failed. The error was: template error while templating string: unexpected '%'. String: {% if ( %} True {% else %} False {% endif %}\n\nThe error appears to have been in '/opt/src/ansible/dom/roles/dom-roles/func/tasks/main.yml': line 4, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n- block:\n    - include_role:\n      ^ here\n"
}```

**Comment by joaocc** [comment]
[copying information from #19085 for convenience]

This happens regardless of:
- static is not explicitly defined
- static is defined to false
- any of the above with "plain" when clause

On the "when" clause, since 2.1, I prefer to always explicit interpolate (this way, I never forget).
However, your comment on the when led me to find something curious...

plain "when"
```
{
    "failed": true,
    "msg": "The conditional check '{' failed. The error was: template error while templating string: unexpected '%'. String: {% if { %} True {% else %} False {% endif %}\n\nThe error appears to have been in '/opt/src/ansible/dom/roles/dom-roles/func/tasks/main.yml': line 7, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - include_role:\n      ^ here\n"
}
```

explicit interpolated "when"
```
{
    "failed": true,
    "msg": "The conditional check '(' failed. The error was: template error while templating string: unexpected '%'. String: {% if ( %} True {% else %} False {% endif %}\n\nThe error appears to have been in '/opt/src/ansible/dom/roles/dom-roles/func/tasks/main.yml': line 7, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - include_role:\n      ^ here\n"
}
```

The messages are different... It seems it is trying to evaluate the initial char, either the parenthesis or the bracket.

Also, if I use the following (false constant), it works.

```

- block:
    - include_role:
        ...
      when: false

```

**Comment by bcoca** [comment]
Can you test the following?

```
- block:
    - include_role:
          name: "ansible-lib--infra/v1/ez123--moodle-moosh"
          private: yes
      when: (zzz_moodle__skip__moosh is not defined) or (not (zzz_moodle__skip__moosh |bool))
      static:  no
```

**Comment by joaocc** [comment]
Same error.

**Comment by bcoca** [comment]
can it be same issue as #18124? try  making it a list

**Comment by joaocc** [comment]
You mean, make the when clauses a list instead of adding them directly in front of the when?

**Comment by paulRbr** [comment]
Just for information, I stubled upon this issue too. 
Using a list seems to work.

```yaml
- name: Install gitlab-ci-runner
  include_role: 
    name: gitlab-runner
  when: my_condition is not defined
```
**fails** with
```
fatal: [10.146.4.77]: FAILED! => {"failed": true, "msg": "The conditional check 'm' failed. The error was: error while evaluating conditional (m): 'm' is undefined\n\nThe error appears to have been in '/tmp/roles/gitlab_ci/tasks/main.yml': line 5, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Install gitlab-ci-runner\n  ^ here\n"}
```
However the following **works** ok:
```yaml
- name: Install gitlab-ci-runner
  include_role: 
    name: gitlab-runner
  when: 
    - my_condition is not defined
```

**Comment by bcoca** [comment]
this is now working in current devel and in 2.3

**Comment by DaveWK** [comment]
broken in 2.4

**Comment by sivel** [comment]
This issue was resolved in debfb798dd15c9cea0ebfdc95e4bce9cc459c039 and subsequently included in the v2.2.1 release.
